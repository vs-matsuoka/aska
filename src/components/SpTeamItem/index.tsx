import Image, { ImageProps } from 'next/future/image';

type SpTeamItemProps = {
  name: string;
  role: string;
  imgSrc: string;
};

function ResponsiveImage(
  props: Omit<ImageProps, 'fill' | 'width' | 'height'> & {
    width: number;
    height: number;
  }
) {
  const { width, height, src, alt, className, ...imageProps } = props;
  const standard = 750;

  return (
    <div
      className={className}
      style={{
        width: `${(width / standard) * 100}vw`,
        height: `${(height / standard) * 100}vw`
      }}
    >
      <Image src={src} alt={alt} className="max-w-none" fill {...imageProps}></Image>
    </div>
  );
}

const SpTeamItem = ({ name, imgSrc, role }: SpTeamItemProps) => (
  // sp:h-[75px] sp:w-[700px] sp:border-[2px]
  <div className="h-[10vw] w-[93.3333333333vw] border-[0.2666666667vw] border-[#ccad70] bg-gradient-to-t from-[#262426] to-[#3a393a]">
    {/* sp:gap-[6px] */}
    <div className="flex gap-[0.8vw]">
      {/* sp:my-[8px] sp:ml-[9px] */}
      <div className="my-[1.0666666667vw] ml-[1.2vw]">
        <div className="relative flex">
          <ResponsiveImage src={imgSrc} alt={name} width={55} height={55} />
        </div>
      </div>
      <div
        // sp:mt-[5px]
        className="mt-[0.6666666667vw]"
        style={{
          fontFamily: 'FOT-UD角ゴ_ラージ Pr6 E',
          lineHeight: '1'
        }}
      >
        {/* sp:text-[20px] */}
        <p className="text-[2.6666666667vw] text-white" data-testid="role">
          {role}
        </p>
        {/* sp:ml-[15px] sp:mt-[3px] sp:text-[36px] */}
        <p className="ml-[2vw] mt-[0.5333333333vw] text-[4.8vw] text-[#CCAD70]" data-testid="name">
          {name}
        </p>
      </div>
      <div className="flex h-auto grow flex-row-reverse">
        {/* sp:mr-[17px] sp:gap-[18px] */}
        <div className="my-auto mr-[2.2666666667vw] flex gap-[2.4vw]">
          {/* TODO: あとで正しいアイコンに置き換え */}
          {/* sp:h-[55px] sp:w-[55px] */}
          <div className="h-[7.3333333333vw] w-[7.3333333333vw] bg-white">plchld1</div>
          <div className="h-[7.3333333333vw] w-[7.3333333333vw] bg-red-500">plchld2</div>
          <div className="h-[7.3333333333vw] w-[7.3333333333vw] bg-blue-500">plchld3</div>
        </div>
      </div>
    </div>
  </div>
);

export default SpTeamItem;
