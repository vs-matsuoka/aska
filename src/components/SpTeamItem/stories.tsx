import { Story, Meta } from '@storybook/react';
import { useEffect } from 'react';
import SpTeamItem from '.';
import team from 'const/team';

export default {
  title: 'SpTeamItem',
  component: SpTeamItem
} as Meta;

export const Default: Story = () => {
  useEffect(() => {
    // FUCK
    window.FONTPLUS.reload();
  });
  return (
    <>
      <SpTeamItem member={team.unnei[0]} />
    </>
  );
};
