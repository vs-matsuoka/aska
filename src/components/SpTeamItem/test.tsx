import { render, screen } from '@testing-library/react';

import SpTeamItem from '.';

describe('<SpTeamItem />', () => {
  it('should render the heading', () => {
    const { container } = render(<SpTeamItem imgSrc="/Team/test.jpg" name="葉月味" role="特番制作" />);

    expect(screen.getByTestId('role')).toBeInTheDocument();
    expect(screen.getByTestId('name')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
