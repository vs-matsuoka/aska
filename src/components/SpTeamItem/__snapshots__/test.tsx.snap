// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpTeamItem /> should render the heading 1`] = `
<div
  class="h-[10vw] w-[93.3333333333vw] border-[0.2666666667vw] border-[#ccad70] bg-gradient-to-t from-[#262426] to-[#3a393a]"
>
  <div
    class="flex gap-[0.8vw]"
  >
    <div
      class="my-[1.0666666667vw] ml-[1.2vw]"
    >
      <div
        class="relative flex"
      >
        <div
          style="width: 7.333333333333333vw; height: 7.333333333333333vw;"
        >
          <img
            alt="那須ピーマン"
            class="max-w-none"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=640&q=75 640w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=750&q=75 750w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=828&q=75 828w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=1080&q=75 1080w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=1200&q=75 1200w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=1920&q=75 1920w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=2048&q=75 2048w, /_next/image?url=%2FIcon%2F%E9%81%8B%E5%96%B6%2F%E9%81%8B%E5%96%B6_icon_%E9%82%A3%E9%A0%88%E3%83%94%E3%83%BC%E3%83%9E%E3%83%B3.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
    <div
      class="mt-[0.6666666667vw]"
      style="font-family: FOT-UD角ゴ_ラージ Pr6 E; line-height: 1;"
    >
      <p
        class="text-[2.6666666667vw] text-white"
        data-testid="role"
      >
        主催
      </p>
      <p
        class="ml-[2vw] mt-[0.5333333333vw] text-[4.8vw] text-[#CCAD70]"
        data-testid="name"
      >
        那須ピーマン
      </p>
    </div>
    <div
      class="flex h-auto grow flex-row-reverse"
    >
      <div
        class="my-auto mr-[2.2666666667vw] flex gap-[2.4vw]"
      >
        <a
          href="https://www.nicovideo.jp/user/57979076"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="relative"
            style="width: 7.333333333333333vw; height: 7.333333333333333vw;"
          >
            <img
              alt="ニコニコ動画"
              class="max-w-none"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=3840&q=75"
              srcset="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=640&q=75 640w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=750&q=75 750w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=828&q=75 828w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=1080&q=75 1080w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=1200&q=75 1200w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=1920&q=75 1920w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=2048&q=75 2048w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Nico.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
        <div
          class="opacity-25"
        >
          <div
            class="relative"
            style="width: 7.333333333333333vw; height: 7.333333333333333vw;"
          >
            <img
              alt="YouTube"
              class="max-w-none"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=3840&q=75"
              srcset="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=640&q=75 640w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=750&q=75 750w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=828&q=75 828w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=1080&q=75 1080w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=1200&q=75 1200w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=1920&q=75 1920w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=2048&q=75 2048w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Youtube.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </div>
        <a
          href="https://twitter.com/nasu_pmen"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="relative"
            style="width: 7.333333333333333vw; height: 7.333333333333333vw;"
          >
            <img
              alt="Twitter"
              class="max-w-none"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=3840&q=75"
              srcset="/_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=640&q=75 640w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=750&q=75 750w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=828&q=75 828w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=1080&q=75 1080w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=1200&q=75 1200w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=1920&q=75 1920w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=2048&q=75 2048w, /_next/image?url=%2FModal%2F22_Modal_pic_SNS_Twitter.png&w=3840&q=75 3840w"
              style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
